hw/ip/edn/data/edn.hjson

```diff
@@ -0,0 +1,264 @@
+            resval: 1
+        },
+      ]
+    },
+    { name: "CTRL",
+      desc: "EDN control register",
+      swaccess: "rw",
+      hwaccess: "hro",
+      fields: [ 
+        {
+            bits: "0",
+            name: "EDN_ENABLE",
+            desc: '''
+                  Setting this bit will enable the EDS module. 
+                  '''
+        },
+        {
+            bits: "1",
+            name: "BCAST_MODE_DIS",
+            desc: '''
+                  Setting this bit will disable the EDN block to self ack the ENDPOINT0 entropy request
+                  bus. This mode for ENDPOINT0 will ignore the request input.
+                  '''
+        },
+        {
+            bits: "2",
+            name: "AUTO_REQ_MODE",
+            desc: '''
+                  Setting this bit will enable the EDN block to automatically send another request to 
+                  CSRNG application interface. It is assuming that the command has been set up ahead of
+                  time, such that when the previous command has returned ack, then a new command will be
+                  send out again without software intervention. It is expected that the generate command
+                  will be repeatedly so that a continuous supply of entropy can be delivered to the
+                  end points.
+                  '''
+        },
+        {
+            bits: "3",
+            name: "BOOT_REQ_DIS",
+            desc: '''
+                  Setting this bit will disable the feature where the EDN block will automatically
+                  send a boot time request to the CSRNG application interface.
+                  The purpose of this feature is to request entropy as fast as possible after reset,
+                  and during chip boot time.
+                  '''
+        },
+      ]
+    },
+    {
+      name: "SUM_STS",
+      desc: "EDS summary status register",
+      swaccess: "rw",
+      hwaccess: "hwo",
+      fields: [
+        { bits: "0",
+          name: "AUTO_REQ_MODE_SM_STS",
+          desc: '''
+                This bit reflects the status of the AUTO_REQ_MODE state machine. After this mode has
+                been turned off, this status bit should be polled to know when the operation is at
+                a completed state, and it is safe to init registers and set this mode again. A value
+                of zero indicates that the operation is completed.
+                '''
+        }
+        { bits: "31",
+          name: "DIAG",
+          desc: "This this an internal generic diagnostic bit."
+        }
+      ]
+    },
+    {
+      name: "SW_CMD_REQ",
+      desc: "EDN csrng app command request register",
+      swaccess: "wo",
+      hwaccess: "hro",
+      hwext: "true",
+      hwqe: "true",
+      regwen: "REGEN",
+      fields: [
+        { bits: "31:0",
+          name: "SW_CMD_REQ",
+          desc: '''
+                Writing this reqister with defined CSRNG commands will initiate all
```

Comment:
is this the register that will encapsulate the `instantiate`, `generate` commands?
I'm just trying to understand how this command relates to the boot scenario where we want fast entropy.

Reply:
Again, added more words on this based on @rswarbrick feedback, but this is for firmware only. For fast boot, this is all driven by the hardware, only after a 0x1 is written to the CRTL reg. Defaults of 0x01 enable the module, and also enable boot mode and broadcast mode.